### PR TITLE
Remove UI divider if speaker has no social links or description

### LINKED ISSUE #2744: UI issue in Speaker Card
**Describe the bug**

Ther is a divider even  in case of no social media links or User description object in the Speaker Card

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to Event/Speakers
2. Speaker Card with no social media link
3. See error

**Expected behaviour**

The header should not be present in case of no social media link

**Screenshots**

Screenshot from 2019-04-19 15-54-42

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Ubuntu 16.04
 - Browser [e.g. chrome, safari] Chrome

**Additional context**

I am working on it

### PR DESCRIPTION
**Pull request by uds5501**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
It ensures removal of the extra divider in speaker item in public section in case no speaker link is present.

#### Changes proposed in this pull request:

- Add hasSocialLinks in component

Fixes #2744

### CODE REVIEW

**Comment by uds5501** [comment]
@CosmicCoder96 @ritikamotwani Please review!

File: app/components/public/speaker-item.js [hunk 1]
```diff
@@ -5,5 +5,13 @@ export default Component.extend({
   classNames  : ['four wide  speaker column'],
   socialLinks : computed(function() {
     return this.get('speaker').getProperties('twitter', 'facebook', 'github', 'linkedin');
+  }),
+  hasSocialLinks: computed(function() {
```
Comment (CosmicCoder96) [inline app/components/public/speaker-item.js]: there is no need for this. `socialLinks` returns an array, if it's size is 0, you can just use `{{if socialLinks}}` to check if there is any social link.

Comment (uds5501) [reply app/components/public/speaker-item.js]: @CosmicCoder96 The `socialLinks` will always be returning a dictionary of 
`{twitter: <null or value>, facebook: <null or value>, github: <null or value> linkedin: <null or value>}` hence the size will always be 4, also, it doesn't handle the case where the user might have some biographical / short description.

File: app/components/public/speaker-item.js [hunk 2]
```diff
@@ -5,5 +5,13 @@ export default Component.extend({
...
+    var currentSpeaker = this.speaker;
+    if (currentSpeaker.twitter || currentSpeaker.facebook || currentSpeaker.github || currentSpeaker.linkedin || currentSpeaker.shortBiography || currentSpeaker.longBiography || currentSpeaker.speakingExperience) {
```
Comment (ritikamotwani) [inline app/components/public/speaker-item.js]: @uds5501 , I guess you should just return this expression. Wouldn't it itself return a true or a false.

Comment (uds5501) [reply app/components/public/speaker-item.js]: @ritikamotwani Good point. making the changes

**Comment by uds5501** [comment]
@ritikamotwani @CosmicCoder96 Please have a look now

**Comment by kushthedude** [comment]
> #### Checklist
> * [x]  I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
> * [x]  My branch is up-to-date with the Upstream `development` branch.
> * [x]  The acceptance, integration, unit tests and linter pass locally with my changes
> * [x]  I have added tests that prove my fix is effective or that my feature works
> * [x]  I have added necessary documentation (if appropriate)
> 
> #### Short description of what this resolves:
> It ensures removal of the extra divider in speaker item in pubic section in case no speaker link is present.
> 
> #### Changes proposed in this pull request:
> * Add hasSocialLinks in component
> 
> Fixes #2744

Correct the typo dude in `Public Section` 😆

**Pull request merged**